nixos/tests/ceph-single-node-bluestore-dmcrypt.nix

```diff
@@ -0,0 +1,270 @@
+import ./make-test-python.nix (
+  { pkgs, lib, ... }:
+
+  let
+    # the single node ipv6 address
+    ip = "2001:db8:ffff::";
+    # the global ceph cluster id
+    cluster = "54465b37-b9d8-4539-a1f9-dd33c75ee45a";
+    # the fsids of OSDs
+    osd-fsid-map = {
+      "0" = "1c1b7ea9-06bf-4d30-9a01-37ac3a0254aa";
+      "1" = "bd5a6f49-69d5-428c-ac25-a99f0c44375c";
+      "2" = "c90de6c7-86c6-41da-9694-e794096dfc5c";
+    };
+
+  in
+  {
+    name = "basic-single-node-ceph-cluster-bluestore-dmcrypt";
+    meta = with pkgs.lib.maintainers; {
+      maintainers = [ nh2 ];
```

Comment:
I'll add you @benaryorg here as well since I'm not using dmcrypt in Ceph myself.